bin/zopen-build

```diff
@@ -651,18 +659,88 @@ install()
     ZOPEN_NAME="${dir}"
     if [ "${ZOPEN_TYPE}x" = "GITx" ]; then
       branch=$(git rev-parse --abbrev-ref HEAD 2>&1 | sed "s/\//./g")
-      ZOPEN_NAME="${dir}.${branch}"
+      ZOPEN_NAME="${dir}-${branch}"
     fi
 
     paxFileName="${ZOPEN_NAME}.${LOG_PFX}.zos.pax.Z"
     if ! runAndLog "pax -w -z -x pax \"-s#${ZOPEN_INSTALL_DIR}/#${ZOPEN_NAME}.${LOG_PFX}.zos/#\" -f \"${paxFileName}\" \"${ZOPEN_INSTALL_DIR}/\""; then
       printError "Could not generate pax \"${paxFileName}\""
     fi
+
+    generateOCI $ZOPEN_NAME
   else
     printHeader "Skipping Install"
   fi
 }
 
+generateOCI()
+{
+  if ! ${publishOCI}; then
+    return 0;
+  fi
+  package=$1
+  name=$(echo $package | cut -d "-" -f 1)
+  version=$(echo $package | cut -d "-" -f 2)
+
+  # Make the version OCI compat
+  case "$version" in
+      *.*.*.*) ;;
+      *.*.*)  version="$version.0";;
+      *.*)  version="$version.0.0";;
+      *) version="1.0.0.0";;
+  esac
+
+  dockerfilecontents="
+FROM scratch
+
+LABEL name=${name}
+LABEL vendor=OSS
+LABEL version=${version}
+LABEL specification=1.0.0"
+
+  cd ${ZOPEN_INSTALL_DIR}
+  rm -f zosadkfile
+  for file in *; do
+    dockerfilecontents="${dockerfilecontents}\nCOPY $file @{CONTAINER_READWRITE}/$file"
+  done
+  mkdir -p .zpm
+  cp .env .zpm/.env
+  dockerfilecontents="${dockerfilecontents}\nCOPY .zpm @{CONTAINER_READWRITE}/.zpm"
+
+  # zosadk does not support gid greater than 2097151
+  gid=$(id | cut -f2 -d' ' | cut -f2 -d'=' | cut -f1 -d'(')
+  gid2=$(id | cut -f3 -d' ' | cut -f2 -d'=' | cut -f2 -d'(' | cut -f1 -d')')
+  if [ $gid -gt 2097151 ] && [ ! -z $gid2 ] ; then
+    # Choose a known GID that is less than 2097151
+    chgrp -R $gid2 *
+  fi
+  
+  echo "$dockerfilecontents" > "zosadkfile"
+  if ! iconv -f IBM-1047 -tISO8859-1 <zosadkfile >.zosadkfileascii || ! chtag -tcISO8859-1 .zosadkfileascii || ! mv .zosadkfileascii zosadkfile; then
+    printError "Unable to make .gitattributes ascii for tarball"
+  fi
+
+  if [ -z $ZOPEN_IMAGE_REGISTRY ]; then
+    printWarning "Environment variable ZOPEN_IMAGE_REGISTRY is needed to push an OCI image"
+    return 4;
+  fi
+  if [ -z $ZOPEN_IMAGE_REGISTRY_ID ]; then
+    printWarning "Environment variable ZOPEN_IMAGE_REGISTRY_ID is needed to push an OCI image"
+    return 4;
+  fi
+  if [ -z $ZOPEN_IMAGE_REGISTRY_KEY_FILE ] || [ ! -r $ZOPEN_IMAGE_REGISTRY_KEY_FILE ]; then
+    printWarning "Environment variable ZOPEN_IMAGE_REGISTRY_KEY_FILE is needed to push an OCI image"
+    return 4;
+  fi
+  if [ -z "$(command -v zosadk)" ]; then
+    printWarning "zosadk is not present on your system"
+    return 4
+  fi
+  zosadk build -t "${name}:${version}" .
```

Comment:
are people able to get access to 'zosadk' ?

Reply:
@IgorTodorovskiIBM

Reply:
I've generalized it so that by default it looks for podman, but it can be overriden